File: archive/count-prefixes-of-a-given-string.js
const countPrefixes = function (words, s) {
  let ans = 0
  words.forEach(word => {
    if (word.length <= s.length) {
      let mem = ''
      for (let i = 0; i < word.length; i++) {
        mem += s[i]
      }

      if (word === mem) {
        ans++
      }
    }
  })

  return ans
};

console.log(countPrefixes(["a", "b", "c", "ab", "bc", "abc"], "abc"))

File: go-archive/count-prefixes-of-a-given-string.go
func countPrefixes(words []string, s string) int {
    ans :=0

    for _, word := range words {
        if (len(word) > len(s)) {continue}
        isOk:=true
        for i:=0; i < len(word); i++ {
            if (word[i] != s[i]) {
                isOk = false
                break
            }
        }
        if isOk {ans++}
    }

    return ans
}